ibis/backends/base/sql/alchemy/datatypes.py

```diff
@@ -30,15 +30,15 @@ def compiles_array(element, compiler, **kw):
 class StructType(sat.UserDefinedType):
     def __init__(
         self,
-        pairs: Iterable[tuple[str, sat.TypeEngine]],
+        pairs: Iterable[tuple[str, sat.TypeEngine]] | Mapping[str, sat.TypeEngine],
```

Comment:
Do we really support iterables here?

Reply:
Maybe not, I'll investigate.

Reply:
Nope, we can remove iterable support here.

Reply:
Done.